### ISSUE TITLE
[BUG] Wrong Device/CPU detection & Using deprecated function & Key needed for demo & Should run local, but uploads model & Very slow Qualcomm cloud

### ISSUE THREAD

**Issue by BrickDesignerNL** [quic/ai-hub-models#122]
**Describe the bug**
- I'm using a Qualcomm X Elite X1E-78-100 with Winodws - but the code things I'm using an Android Phone: Samsung Galaxy S23 Ultra.
- Next to that function 'resume_download' is deprecated in favor of 'resume'
- Also I needed a Qualcomm API key, while I wanted to run a local demo. - Feel strange, can that be avoided in future releases?
- It uploads a model, while I expected that it would run locally?
- Upload extreme slow

**To Reproduce**
py -m qai_hub_models.models.stable_diffusion_v2_1_quantized.demo

**Expected behavior**
Optimizing for X Elite instead of phone.
100% local, offline working
No errors ;)

**Stack trace**

> RuntimeError: Stable-diffusion on-device demo requires access to QAI-Hub.
> Please visit https://aihub.qualcomm.com/ and sign-up.

...

> pip install qai-hub
> qai-hub configure --api_token  ***
> py -m qai_hub_models.models.stable_diffusion_v2_1_quantized.demo

...

> C:\Users\<user_name>\python\qai_hub_models_env\Lib\site-packages\huggingface_hub\file_download.py:1132: FutureWarning: `resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`.
>   warnings.warn(

...

> ** Performing image generation on-device(Samsung Galaxy S23 Ultra) with Stable Diffusion **

...

> Prompt: spectacular view of northern lights from Alaska
> Number of steps: 5
> Guidance scale: 7.5
> Seed: 0
> Note: This reference demo uses significant amounts of memory and may take 4-5 minutes to run ** per step **.

> ----------------------------------------------------------------------------------------------------

> Downloading model assets

> -----------------------------------

> Uploading model assets on QAI-Hub

> -----------------------------------
> Uploading model:  81%|██████████████████████████████████████████████████▉            | 305M/378M [15:24<03:13, 392kB/s]

**Host configuration:**
 - OS and version: Windows
 - Python:  3.12.7 (x86)
 - QAI-Hub-Models version: stable_diffusion_v2_1_quantized
 - Internet connection: 4Gbit/s - so it's not me that's slow

**Additional context**
It doesn't feel like the model runs locally, and even with 4Gbit/s internet at home, the upload is extreme slow.
So at the moment running Stable Diffusion is far from what I expect on a CoPilot+ PC.
And until now it doesn't utilize the GPU nor NPU, it just (slowly) uploads to the Qualcomm cloud.

https://huggingface.co/qualcomm/Stable-Diffusion-v2.1#demo-on-device says:

> #Demo on-device
> The package contains a simple end-to-end demo that downloads pre-trained weights and runs this model on a sample input.
> 
> `python -m qai_hub_models.models.stable_diffusion_v2_1_quantized.demo`
> 
> The above demo runs a reference implementation of pre-processing, model inference, and post processing.

And it says:
> This repository provides scripts to run Stable-Diffusion-v2.1 on Qualcomm® devices.

But it's uploading data instead of downloading and not it's not running on-device.

**Comment by kory** [comment]
> Upload extreme slow

Unfortunately, we can't help with this one. You're uploading directly to Amazon's S3 servers. Slow uploads are caused by a slow connection or issues on their end.

**Comment by kory** [comment]
> FutureWarning: resume_download is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use force_download=True.

We appreciate the report here. This is an issue caused by a downstream dependency (diffusers). We'll try to update the package to remove the deprecation warning.

**Comment by BrickDesignerNL** [comment]
@kory The website itself contains 3 binaries, 
https://huggingface.co/qualcomm/Stable-Diffusion-v2.1/tree/main

Can't those be used on my X Elite?
I actually bought it to be able to run those Diffusion models locally.
And the commercial materiaal actually promotes that it runs fast 

So I was under the impression that it would be as easy as on my PC with Nvidia card to run the models on the laptop.
But until now I only got ollama running locally.

**Comment by BrickDesignerNL** [comment]
> > Upload extreme slow
> 
> Unfortunately, we can't help with this one. You're uploading directly to Amazon's S3 servers. Slow uploads are caused by a slow connection or issues on their end.

Typically S3 is very fast in my region (easy over 800Mbit/s).
You can choose to use https://docs.aws.amazon.com/AmazonS3/latest/userguide/transfer-acceleration.html that would speed up your S3 for everyone in the world.

**Comment by BrickDesignerNL** [comment]
> > Also I needed a Qualcomm API key, while I wanted to run a local demo. - Feel strange, can that be avoided in future releases?
> 
> This is not avoidable if you want to run an on-device demo. We need to provision a device in the cloud for your model, and that requires authentication.

For me, on-device means it runs on my local device.
I can even choose my CPU, so it should be able to run the model on my device.

Can you rephrase it like "Cloud hosted device" and provide instructions how to run it on my own device if I have the same device as Qualcomm has on the cloud.
Feels very strange that I can use a CPU equal to mine in the cloud to run the model, but can't run the same model on my device with the same CPU.

Why would anyone want that?

**Comment by BrickDesignerNL** [comment]
For me also "Physical Device" sounds like it's the device I own.

I bought this X Elite computer because of this repo the sounds like the models can run on my device.

**Comment by kory** [comment]
> @kory The website itself contains 3 binaries, https://huggingface.co/qualcomm/Stable-Diffusion-v2.1/tree/main
> 
> Can't those be used on my X Elite? I actually bought it to be able to run those Diffusion models locally. And the commercial materiaal actually promotes that it runs fast
> 
> So I was under the impression that it would be as easy as on my PC with Nvidia card to run the models on the laptop. But until now I only got ollama running locally.

That's correct--these models run on the X Elite. We don't have an end-to-end local device example for stable diffusion at this time, however. Please join our [Slack](https://aihub.qualcomm.com/community/slack) and we'd be happy to get you going on that. We actively monitor Mon-Fri.  

We're very aware there is a gap -- "how do I use this asset?" after downloading a compiled model. It's an active effort to document this next step and provide more samples for on-device execution.

> Typically S3 is very fast in my region (easy over 800Mbit/s).
You can choose to use https://docs.aws.amazon.com/AmazonS3/latest/userguide/transfer-acceleration.html that would speed up your S3 for everyone in the world.

What is your region? I'll forward this suggestion to the appropriate teams to see if we are using this already. If not, it's definitely a great suggestion, and appreciated :)

> an you rephrase it like "Cloud hosted device" and provide instructions how to run it on my own device if I have the same device as Qualcomm has on the cloud.

I understand the confusion, and appreciate the feedback. We'll definitely discuss how to improve the messaging here.

> Feels very strange that I can use a CPU equal to mine in the cloud to run the model, but can't run the same model on my device with the same CPU. Why would anyone want that?

This repository is designed to run on any computer that supports python. Many developers use computers that don't have Qualcomm chipsets when developing models for Qualcomm platforms.

We built the AI Hub cloud service to allow compilation & direct execution on any Qualcomm device in the cloud. That way, you don't need to set up your own device to test and validate your model inference time & accuracy. The X Elite in our view is one of many devices to support in this context.

Applications that target specifically the X Elite are important, and are included in our ["applications" repository](https://github.com/quic/ai-hub-apps/tree/main/apps/windows), not our "models" repository. Our long-term goal is to have an on-device application for every model.
